test/test_util/test_trials_callback.py

```diff
@@ -0,0 +1,92 @@
+import os
+import shutil
+import unittest
+
+import pandas as pd
+
+from smac.callbacks import IncorporateRunResultCallback
+from smac.optimizer.smbo import SMBO
+from smac.runhistory.runhistory import RunInfo, RunValue
+from smac.tae.base import StatusType
+
+import autosklearn.pipeline.util as putil
+from autosklearn.classification import AutoSklearnClassifier
+
+
+class AutoMLTrialsCallBack(IncorporateRunResultCallback):
+
+    def __init__(self, output_dir):
+        self.trials_num = 1
+        self.output_dir = output_dir
+        with open(os.path.join(self.output_dir, "trials.csv"), "w") as fp:
+            fp.write("TrialNo, StartTime, EndTime, Status, TrainLoss, ValidLoss, TestLoss, Classifier")
+
+    def __call__(
+            self, smbo: 'SMBO',
+            run_info: RunInfo,
+            result: RunValue,
+            time_left: float,
+    ) -> None:
+        train_loss, valid_loss, test_loss = None, None, None
+        trial_start_time = result.starttime
+        trial_end_time = result.endtime
+        trial_status = result.status.name
+        if trial_status == StatusType.SUCCESS.name:
+            train_loss = result.additional_info.get('train_loss')
+            valid_loss = result.cost
+            test_loss = result.additional_info.get('test_loss')
+        trial_classifier = run_info.config.get_dictionary()['classifier:__choice__']
+        with open(os.path.join(self.output_dir, "trials.csv"), "a+") as fp:
+            fp.write(f"\n {self.trials_num}, {trial_start_time}, {trial_end_time}, {trial_status}, "
+                     f"{train_loss}, {valid_loss}, {test_loss}, {trial_classifier}")
+        self.trials_num += 1
+
+
+class VerifyTrialsCallBack(unittest.TestCase):
+
+    def test_trials_callback_execution(self):
+        base_dir = "./tmp"
```

Comment:
Could you please use python's built-in `tempdir` module?

Reply:
Hi @mfeurer, it makes sense to allow for newly added callbacks in SMAC. Modified the code to support it.